### PR TITLE
connecting etl pipeline scripts

### LINKED ISSUE #18: Create an etl.py script that connects the whole pipeline
# Description
A full ETL pipeline that connects the `extract.py`, `transform.py`, and `load.py`, so that it can be run in one command 

Outcome
- `etl.py` : Python script that imports the relevant functions from `extract.py`, `transform.py`, and `load.py` to form the whole pipeline, ready to be dockerised.

#User Story
As an Engineer, when I run the etl.py script, I want the data to be extracted from an api (by specifying website+product url), load to a dataframe, cleaned and saved to a new dataframe, then uploaded to the RDS database.

### PR DESCRIPTION
**Pull request by gem09lo**
This connects `extract.py`, `transform.py`, and `load.py` to form the ETL pipeline so it can be dockerised and hosted on the cloud.
The email script is yet to be added. Not ready to be merged yet.

This (partially) closes #18

### CODE REVIEW

File: pipeline/email_notifier.py [hunk 1]
```diff
@@ -14,12 +14,12 @@ def get_ses_client() -> boto3.client:
     """Returns the boto3 SES client to send emails with."""
 
     aws_access_key = ENV.get("AWS_ACCESS_KEY_ID")
-    aws_secret_access_key = ENV.get("AWS_ACCESS_SECRET_KEY")
+    aws_secret_access_key = ENV.get("AWS_SECRET_ACCESS_KEY")
```
Comment (indiah444) [inline pipeline/email_notifier.py:17]: Thank you so much for spotting this error, can't believe I added this in :')

File: pipeline/load.py [hunk 2]
```diff
@@ -50,12 +50,8 @@ def load_price_changes(products_data: list[dict], conn: connection) -> None:
         logging.error("Load error during commit: %s", commit_error)
 
 
-def main_load():
+def main_load(cleaned_data: list[dict]) -> None:
```
Comment (indiah444) [inline pipeline/load.py:53]: Great that you've moved away from hard coding in test data and allow for inputs!

File: pipeline/test_email_notifier.py
Comment (indiah444) [inline pipeline/test_email_notifier.py:1]: Again, thank you for correcting my mistake :')

File: test_database.sql [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
-    ('Jane', 'Smith', 'jane.smith@example.com', 'securepass'),
-    ('Alice', 'Johnson', 'alice.j@example.com', 'mypassword'),
-    ('Bob', 'Williams', 'bob.w@example.com', 'pass1234'),
-    ('John', 'Smith', 'john_s@example.com', 'abcdef');
+    -- ('John', 'Doe', 'john.doe@example.com', 'password123'),
+    -- ('Jane', 'Smith', 'jane.smith@example.com', 'securepass'),
+    -- ('Alice', 'Johnson', 'alice.j@example.com', 'mypassword'),
+    -- ('Bob', 'Williams', 'bob.w@example.com', 'pass1234'),
+    -- ('John', 'Smith', 'john_s@example.com', 'abcdef'),
+    ('John', 'Doe', 'chris.owen@sigmalabs.co.uk', 'password123'),
```
Comment (indiah444) [inline test_database.sql:8]: Good test data, will allow us to make sure the pipeline works in the cloud :)

File: pipeline/etl.py
Comment (indiah444) [inline pipeline/etl.py:1]: Great file, nice and clean and connects the pipeline well!

**Review by indiah444 — APPROVED** [review]
Thank you for spotting and changing a mistake I made in my file, it's great that the pipeline is now working!

File: pipeline/email_notifier.py (hunk 1 above)
Comment (gem09lo) [reply pipeline/email_notifier.py:17]: Thanks!

**Pull request merged**